### PR TITLE
Add dev docs for creating a new add-on in zap-extensions

### PR DESCRIPTION
**Pull request by psiinon**
Screenshot_2021-03-22 OWASP ZAP – Creating a New Add-on in zap-extensions

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+```bash
+mv addOns/myaddon/simpleexample.gradle.kts addOns/myaddon/myaddon.gradle.kts
+```
+On Windows this can be done from the command line using:
+```bash
+rename addOns\myaddon\simpleexample.gradle.kts addOns\myaddon\myaddon.gradle.kts
+```
+Edit the new file `myaddon.gradle.kts` and change the description and name:
+```
+version = "2"
```
Comment (thc202) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:54]: `0.0.1`?

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+description = "A new description."
+
+zapAddOn {
+    addOnName.set("My AddOn")
+    zapVersion.set("2.9.0")
```
Comment (thc202) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:59]: `2.10.0`

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	<comment>Project myaddon created by Buildship.</comment>
+	<projects>
+	</projects>
+```
+
+### settings.gradle.kts
+
+Edit the file `settings.gradle.kts` and add `myaddon` into the `addOns` list.
+
+Refresh the gradle project in your IDE.
```
Comment (thc202) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:83]: Gradle

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+- guide
+- tutorial
+type: page
+date: "2021-03-19"
+---
+
+ZAP has a plugin architecture and new functionality is implemented via add-ons.
+
+Add-ons can be defined in any repository but most of the ones that the ZAP core team maintains live in 
+[zap-extensions](https://github.com/zaproxy/zap-extensions/).
```
Comment (thc202) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:14]: Should this be more clear that's for add-ons that are intended to be permanently added to the `zap-extensions` repo?

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    addOnName.set("My AddOn")
+    zapVersion.set("2.9.0")
+
+    manifest {
+        author.set("ZAP Dev Team")
+    }
+}
+```
+
+### .project
```
Comment (thc202) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:67]: Remove? Eclipse specific and Gradle refresh should recreate it.

**Comment by psiinon** [comment]
Hopefully fixed all of those.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by psiinon** [comment]
Part of https://github.com/zaproxy/zaproxy-website/issues/366

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+If you are planning on maintaining your add-on yourself then you should create it in another repository.
+
+This guide will explain how you can quickly create a brand new add-on in this repository that you can then customise as you want.
+
+## Preparation
+You will need to have followed the [Quick Start Guide to Building ZAP](../quick-start-build/) and have imported the ZAP code into your favourite IDE.
+
+For the command line instructions given below you should be in the `zap-extensions` directory.
+
+## Copy the Simple Example
```
Comment (kingthorin) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:26]:
```suggestion
## Copy the Simple Example Add-on
```

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+    manifest {
+        author.set("ZAP Dev Team")
+    }
+}
+```
+
+### settings.gradle.kts
+
+Edit the file `settings.gradle.kts` and add `myaddon` into the `addOns` list.
```
Comment (kingthorin) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:72]: Do we need to provide the full path? (Or a leading slash) [I didn't go looking]

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+Refresh the Gradle project in your IDE.
+
+## Update the Source Code
+
+These are the minimum changes needed to get your new add-on working. 
+
+### Message.properties
+
+Edit the file `addOns/myaddon/src/main/resources/org/zaproxy/zap/extension/simpleexample/resources/Messages.properties` 
+and change all of the instances of `simpleexample` in the keys to `myaddon` as well as changing the `Simple Example` strings to something like `My Add-on`.
```
Comment (kingthorin) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:83]: What about the second `simpleexample` in the path?

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+Your new add-on should now be available in ZAP. It will add a new menu item and a new tab in the bottom window - you will need to click on the green plus to see it.
+
+![ZAP Screenshot with My Add-on](/img/docs/developer/myaddon.png)
+
+## Next Steps
+
+You should update the following files before you make too many other changes:
+
+### Package name
```
Comment (kingthorin) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:137]:
```suggestion
### Package Name
```

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+Change the package from `org/zaproxy/zap/extension/simpleexample` to something like `org/zaproxy/addon/myaddon/`
+You can change the package name (and class names in the following section) by just editing the files but in general your IDE is likely to make this easier.
+
+### Class names
```
Comment (kingthorin) [inline site/content/docs/developer/creating-new-addon-in-zap-extensions.md:142]:
```suggestion
### Class Names
```

**Review by kingthorin — CHANGES_REQUESTED** [review]
@thc202 should we move simpleexample to the newer `addon` paths (vs. `extension`)? [Like the retire add-on? https://github.com/zaproxy/zap-extensions/tree/master/addOns/retire/src/main/java/org/zaproxy/addon/retire)

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md (hunk 7 above)
Comment (psiinon) [reply site/content/docs/developer/creating-new-addon-in-zap-extensions.md:72]: Its in the cwd. I disnt want to add a leading slash as that would take us to root ;)

File: site/content/docs/developer/creating-new-addon-in-zap-extensions.md (hunk 8 above)
Comment (psiinon) [reply site/content/docs/developer/creating-new-addon-in-zap-extensions.md:83]: that needs to stay - but I do mention that later

**Comment by psiinon** [comment]
Updated the titles :)

**Comment by thc202** [comment]
I'm +1 to move it to the new package.

**Comment by psiinon** [comment]
Thats mentioned in the "Next Steps". I dont want to include it before as its a complication which isnt needed.
The rest can be done very easily from the cmd line, the package rename is much easier in your IDE.

**Review by kingthorin — APPROVED** [review]

**Pull request merged**